csharp/compatibility_tests/v3.0.0/test.sh

```diff
@@ -0,0 +1,102 @@
+#!/bin/bash
+
+function run_test() {
+  # Generate test proto files.
+  ./protoc_1 -Iprotos/src -I../../../src/ --csharp_out=src/Google.Protobuf.Test \
+    --csharp_opt=base_namespace=Google.Protobuf \
+    protos/src/google/protobuf/map_unittest_proto3.proto \
+    protos/src/google/protobuf/unittest_import_proto3.proto \
+    protos/src/google/protobuf/unittest_import_public_proto3.proto \
+    protos/src/google/protobuf/unittest_well_known_types.proto
+
+  ./protoc_1 -Iprotos/csharp --csharp_out=src/Google.Protobuf.Test \
+    --csharp_opt=base_namespace=UnitTest.Issues \
+    protos/csharp/protos/unittest_issues.proto
+
+  ./protoc_2 -Iprotos/src --csharp_out=src/Google.Protobuf.Test \
```

Comment:
No. 
The test separate proto files into two sets. Set 2 proto files may import some of proto files in set 1.
3 cases are tested:
Case 1:
protobuf runtime: new runtime
both .proto files set 1 and set 2 use v3.0.0 protoc

Case 2:
protobuf runtime: new runtime
.proto files set 1 uses new version protoc
.proto files set 2 uses v3.0.0 protoc

Case 3:
protobuf runtime : new runtime
.proto files set 1 uses v3.0.0 protoc
.proto files set 2 uses new version protoc